src/main/java/org/opensearch/ad/ml/ModelManager.java

```diff
@@ -171,34 +171,87 @@ public void getRcfResult(String detectorId, String modelId, double[] point, Acti
         }
     }
 
-    private void getRcfResult(ModelState<ThresholdedRandomCutForest> modelState, double[] point, ActionListener<RcfResult> listener) {
+    private void getTRcfResult(
+        ModelState<ThresholdedRandomCutForest> modelState,
+        double[] point,
+        ActionListener<ThresholdingResult> listener
+    ) {
         modelState.setLastUsedTime(clock.instant());
 
-        ThresholdedRandomCutForest rcf = modelState.getModel();
-        AnomalyDescriptor result = rcf.process(point, 0);
-        double score = result.getRcfScore();
-        double confidence = result.getDataConfidence();
-        int forestSize = result.getForestSize();
-        double grade = result.getAnomalyGrade();
+        ThresholdedRandomCutForest trcf = modelState.getModel();
+        try {
+            AnomalyDescriptor result = trcf.process(point, 0);
+            double[] attribution = normalizeAttribution(trcf.getForest(), result.getCurrentTimeAttribution());
+            listener
+                .onResponse(
+                    new ThresholdingResult(
+                        result.getAnomalyGrade(),
+                        result.getDataConfidence(),
+                        result.getRcfScore(),
+                        result.getTotalUpdates(),
+                        result.isStartOfAnomaly(),
+                        result.isInHighScoreRegion(),
+                        result.getRelativeIndex(),
+                        attribution,
+                        result.getOldValues(),
+                        result.getExpectedValuesList(),
+                        result.getThreshold(),
+                        result.getForestSize()
+                    )
+                );
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
 
+    /**
+     * normalize total attribution to 1
+     *
+     * @param forest rcf accessor
+     * @param rawAttribution raw attribution scores
+     *
+     * @return normalized attribution
+     */
+    public double[] normalizeAttribution(RandomCutForest forest, double[] rawAttribution) {
```

Comment:
This may be covered somewhere else, but want to confirm: how is it guaranteed that the array of attribution values is in the same order as the `List` of features for the detector?

Reply:
same question for expected value.